share/revng/test/tests/analysis/Decompilation/x86-64/pretty-ints.c

```diff
@@ -0,0 +1,57 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <stdint.h>
+#include <stddef.h>
+
+#define WEAK __attribute__((weak))
+
+WEAK uint64_t shift_left(int64_t amount) {
+  return 0xABCDEF1234567890ULL << amount;
+}
+
+WEAK uint64_t logical_shift_right(int64_t amount) {
+  return 0xABCDEF1234567890ULL >> amount;
+}
+
+WEAK int64_t arithmetic_shift_right(int64_t amount) {
+  return 0xABCDEF1234567890LL >> amount;
+}
+
+WEAK uint64_t bitwise_and(int64_t arg) {
+  return 0xABCDEF1234567890ULL & arg;
+}
+
+WEAK uint64_t bitwise_or(int64_t arg) {
+  return 0xABCDEF1234567890ULL | arg;
+}
+
+WEAK uint64_t bitwise_xor(int64_t arg) {
+  return 0xABCDEF1234567890ULL ^ arg;
+}
+
+WEAK int8_t data;
+
+WEAK int8_t write_and_get_char() {
+  return data = 'X';
+}
+
+WEAK int puts(const char *ptr) {
+  data = *ptr;
+}
+
+WEAK
+int main(int argc, const char **argv) {
+  puts(NULL);
+  return shift_left(argc) + logical_shift_right(argc)
+         + arithmetic_shift_right(argc) + bitwise_or(argc) + bitwise_and(argc)
+         + bitwise_xor(argc) + write_and_get_char();
+}
+
+WEAK int argc;
+WEAK const char **argv;
+
+void _start() {
+  main(argc, argv);
+}
```

Comment:
Usually you either have `_start` or `main`. Having both is a bit confusing.

Drop `main` and use the global directly?

Same in `segments-and-sections.c`.

Reply:
I still see `main`

Reply:
Now should be ok.